flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/declarative/DeclarativeScheduler.java

```diff
@@ -414,6 +421,10 @@ public JobDetails requestJobDetails() {
     @Override
     public KvStateLocation requestKvStateLocation(JobID jobId, String registrationName)
             throws UnknownKvStateLocation, FlinkJobNotFoundException {
+        if (!this.jobInformation.getJobID().equals(jobId)) {
+            throw new FlinkJobNotFoundException(jobId);
```

Comment:
may need to extend this to other KvState methods as well; need to double-check which methods are used by tests.

Reply:
A more fundamental question is why this method even accepts a job ID in the first place.

Reply:
I think we are doing this check already in `KvStateHandler`.